main.go

```diff
@@ -8,6 +8,11 @@ import (
 	log "github.com/Sirupsen/logrus"
 	"github.com/rancherio/go-machine-service/events"
 	"github.com/rancherio/go-machine-service/handlers"
+	_ "github.com/rancherio/go-machine-service/handlers/providers/amazonec2"
```

Comment:
That is not needed in this scenario. It would be needed if there was any state associated with the structs and we need to initialize these fields. In this case, we do not need to initialize the providers. Hence, i'm using init(), instead of a New(fields...). I cannot think of a scenario where that would be needed, that is why i designed it this way.